server/channels/api4/job_test.go

```diff
@@ -336,3 +368,61 @@ func TestCancelJob(t *testing.T) {
 	require.Error(t, err)
 	CheckNotFoundStatus(t, resp)
 }
+
+func TestUpdateJobStatus(t *testing.T) {
+	th := Setup(t)
+	defer th.TearDown()
+
+	jobType := model.JobTypeDataRetention
+	jobs := []*model.Job{
+		{
+			Id:     model.NewId(),
+			Type:   jobType,
+			Status: model.JobStatusPending,
+		},
+		{
+			Id:     model.NewId(),
+			Type:   jobType,
+			Status: model.JobStatusInProgress,
+		},
+		{
+			Id:     model.NewId(),
+			Type:   jobType,
+			Status: model.JobStatusSuccess,
+		},
+	}
+
+	for _, job := range jobs {
+		_, err := th.App.Srv().Store().Job().Save(job)
+		require.NoError(t, err)
+		defer th.App.Srv().Store().Job().Delete(job.Id)
+	}
+
+	defaultRolePermissions := th.SaveDefaultRolePermissions()
+	defer func() {
+		th.RestoreDefaultRolePermissions(defaultRolePermissions)
+	}()
+	th.AddPermissionToRole(model.PermissionManageDataRetentionJob.Id, model.SystemAdminRoleId)
```

Comment:
That is surprising. With the correct migrations in place, I expect system admins to manage jobs without additional changes.

Reply:
That was needed before I made the proper migration change. Removed it now